daemon/images/image_import.go

```diff
@@ -4,102 +4,48 @@ import (
 	"context"
 	"encoding/json"
 	"io"
-	"net/http"
-	"net/url"
-	"strings"
 	"time"
 
 	"github.com/containerd/containerd/platforms"
 	"github.com/docker/distribution/reference"
 	"github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/builder/dockerfile"
-	"github.com/docker/docker/builder/remotecontext"
 	"github.com/docker/docker/dockerversion"
 	"github.com/docker/docker/errdefs"
 	"github.com/docker/docker/image"
 	"github.com/docker/docker/layer"
 	"github.com/docker/docker/pkg/archive"
-	"github.com/docker/docker/pkg/progress"
-	"github.com/docker/docker/pkg/streamformatter"
 	"github.com/docker/docker/pkg/system"
 	specs "github.com/opencontainers/image-spec/specs-go/v1"
-	"github.com/pkg/errors"
 )
 
-// ImportImage imports an image, getting the archived layer data either from
-// inConfig (if src is "-"), or from a URI specified in src. Progress output is
-// written to outStream. Repository and tag names can optionally be given in
-// the repo and tag arguments, respectively.
-func (i *ImageService) ImportImage(ctx context.Context, src string, repository string, platform *specs.Platform, tag string, msg string, inConfig io.ReadCloser, outStream io.Writer, changes []string) error {
-	var (
-		rc     io.ReadCloser
-		resp   *http.Response
-		newRef reference.Named
-	)
-
-	if repository != "" {
-		var err error
-		newRef, err = reference.ParseNormalizedNamed(repository)
-		if err != nil {
-			return errdefs.InvalidParameter(err)
-		}
-		if _, isCanonical := newRef.(reference.Canonical); isCanonical {
-			return errdefs.InvalidParameter(errors.New("cannot import digest reference"))
-		}
-
-		if tag != "" {
-			newRef, err = reference.WithTag(newRef, tag)
-			if err != nil {
-				return errdefs.InvalidParameter(err)
-			}
-		}
-	}
-
-	// Normalize platform - default to the operating system and architecture if not supplied.
+// ImportImage imports an image, getting the archived layer data from layerReader.
+// Layer archive is uncompressed and recompressed as gzip for internal storage.
```

Comment:
That's not quite true: the uncompressed layer archive is passed to the graphdriver, not recompressed. And as far as I am aware, none of our graph drivers store compressed archives of layers anywhere.

Reply:
Thanks, fixed!